### PR TITLE
Loggestd

### PR DESCRIPTION
**Pull request by r-darwish**

### CODE REVIEW

File: loggestd/src/usage_monitor.rs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+pub struct UsageMonitor {
+    interval: Interval,
+}
+
+fn get_entries_with_metadata(directory: &Path) -> Result<Vec<(DirEntry, Option<Metadata>)>, io::Error> {
+    let readdir = fs::read_dir(directory)?;
+
+    let mut result: Vec<(DirEntry, Option<Metadata>)> =
+        Vec::with_capacity(dbg!(readdir.size_hint()).1.unwrap_or(0));
```
Comment (drrlvn) [inline loggestd/src/usage_monitor.rs]: Remove `dbg!()`.

File: loggestd/src/usage_monitor.rs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    } else {
+        debug!("No need for GC");
+    }
+
+    Ok(())
+}
+
+impl Default for UsageMonitor {
+    fn default() -> UsageMonitor {
+        let interval = Interval::new(Instant::now(), Duration::from_secs(10));
```
Comment (drrlvn) [inline loggestd/src/usage_monitor.rs]: Should be 1 minute.

**Comment by r-darwish** [comment]
All discussions are resolved

File: loggestd/src/log_file.rs [hunk 3]
```diff
@@ -5,58 +5,55 @@ use std::io;
...
-        let new_extension = filename
-            .extension()
-            .map(|e| {
-                let mut e = e.to_os_string();
-                e.push(".1");
-                e
-            })
-            .unwrap_or_else(|| "1".into());
-        filename.set_extension(new_extension);
+fn generate_filename(base_name: &Path, index: usize) -> PathBuf {
```
Comment (drrlvn) [inline loggestd/src/log_file.rs]: Should be a method and use `self.base_filename`.

**Pull request merged**